src/engine/GameObjects/MapNode.cxx

```diff
@@ -205,50 +205,74 @@ bool MapNode::isPlacementAllowed(const std::string &newTileID) const
-      }
-      else if ((isLayerOccupied(Layer::BUILDINGS) && (m_mapNodeData[Layer::BUILDINGS].tileData->category != "Flora")) ||
-               isLayerOccupied(Layer::WATER) || !isPlacableOnSlope(newTileID))
-      {
+      if ((isLayerOccupied(Layer::BUILDINGS) && (m_mapNodeData[Layer::BUILDINGS].tileData->category != "Flora")) ||
+          isLayerOccupied(Layer::WATER) || !isPlacableOnSlope(newTileID))
+      { // roads cannot be placed:
+        // - on buildings that are not category flora.
+        // - on water
+        // - on slopetiles that don't have a tileID
         return false;
       }
+      return true;
+    case Layer::GROUND_DECORATION:
+      if (m_mapNodeData[Layer::GROUND_DECORATION].tileData || m_mapNodeData[Layer::BUILDINGS].tileData)
+      { // allow placement of ground decoration on existing ground decoration and on buildings.
+        return true;
+      }
+      break;
     case Layer::BUILDINGS:
-      if (isLayerOccupied(Layer::ROAD))
-      {
-        return false;
+      TileData *tileDataBuildings = m_mapNodeData[Layer::BUILDINGS].tileData;
+      if (tileDataBuildings && tileDataBuildings->isOverPlacable)
+      { // buildings with overplacable flag
+        return true;
       }
-      if (m_mapNodeData[layer].tileData && m_mapNodeData[layer].tileData->tileType == +TileType::DEFAULT &&
-          !m_mapNodeData[layer].tileData->isOverPlacable)
-      {
+      if (isLayerOccupied(Layer::ROAD))
+      { // buildings cannot be placed on water
         return false;
       }
+      break;
     }
-    //this is a water tile and placeOnWater has not been set to true, building is not permitted. Also disallow placing of water tiles on non water tiles
-    if (tileData->tileType != +TileType::WATER &&
-        (isLayerOccupied(Layer::WATER) && !tileData->placeOnWater || !isLayerOccupied(Layer::WATER) && tileData->placeOnWater))
+
+    // checks for all layers:
+    if (isLayerOccupied(Layer::WATER) && tileData->tileType != +TileType::WATER && !tileData->placeOnWater)
+    // Disallow placement on water for tiles that are:
+    // not of tiletype water
+    // not flag placeOnWater enabled
     {
       return false;
     }
 
-    // check if the current tile has the property overplacable set or if it's of the same tile ID for certain TileTypes only (not DEFAULT)
-    if (m_mapNodeData[layer].tileData &&
-        (m_mapNodeData[layer].tileData->tileType == +TileType::GROUNDDECORATION ||
-         m_mapNodeData[layer].tileData->isOverPlacable ||
-         (m_mapNodeData[layer].tileData->tileType != +TileType::DEFAULT && m_mapNodeData[layer].tileID == newTileID)))
+    if (!isLayerOccupied(Layer::WATER) && !tileData->placeOnGround)
+    // Disallow placement on ground (meaning a tile that is not water) for tiles that have:
+    // not flag placeOnGround enabled
     {
+      return false;
+    }
+
+    if(!isPlacableOnSlope(newTileID))
+    { // Check if a tile can be placed on a slop tile
+      return false;
+    }
```

Comment:
I am not sure about this one. What is the meaning here?

Reply:
Not all tiles can be placed on slopes, only certain tiles have a frame for slope.
I've update the comment
// Check if a tile has slope frames and therefore can be placed on a node with a slope